src/components/visualizations/chart/chartOptionsBuilder.ts

```diff
@@ -1781,8 +1811,14 @@ export function getChartOptions(
         };
     }
 
-    const tooltipFn = isTreemap(type) ? generateTooltipTreemapFn(viewByAttribute, stackByAttribute, config) :
-        generateTooltipFn(viewByAttribute, type, config);
+    let tooltipFn;
+    if (isTreemap(type)) {
+        tooltipFn = generateTooltipTreemapFn(viewByAttribute, stackByAttribute, config);
+    } else if (isViewByTwoAttributes) {
+        tooltipFn = generateTooltipForTwoAttributesFn(viewByAttribute, viewByParentAttribute, config);
+    } else {
+        tooltipFn = generateTooltipFn(viewByAttribute, type, config);
+    }
```

Comment:
The original `getChartOptions` function is quite long already. Would it be possible to do the following?

1) Rename `generateTooltipTreemapFn` to `buildTooltipTreemapFactory`
2) Rename `generateTooltipForTwoAttributesFn` to `buildTooltipForTwoAttributesFactory`
3) Rename `generateTooltipFn` to buildTooltipFactory`
4) Extract whole changed block of code to a new private function `getTooltipFactory(viewByAttribute, viewByParentAttribute, stackByAttribute, config)` method where you can just extract `type` from config, have your if/else if/else blocks that have three return statements that just returns all your three factory builders and then use this function either directly as value of `tooltip:` key in the built object or assign it into the `tooltipFactory` constant.

Reply:
I will do this in a seperated commit

Reply:
the refactor is here https://github.com/gooddata/gooddata-react-components/pull/881/commits/84cb622c7636dc1aef81e96fe3f62cc4995f1aac